WeScan/Edit/ZoomGestureController.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  ZoomGestureController.swift
+//  WeScan
+//
+//  Created by Bobo on 5/31/18.
+//  Copyright © 2018 WeTransfer. All rights reserved.
+//
+
+import Foundation
+import AVFoundation
+
+final class ZoomGestureController {
+    
+    private let image: UIImage
+    private let quadView: QuadrilateralView
+    
+    init(image: UIImage, quadView: QuadrilateralView) {
+        self.image = image
+        self.quadView = quadView
+    }
+    
+    private var previousPanPosition: CGPoint?
+    private var closestCorner: CornerPosition?
+    
+    @objc func handle(pan: UIGestureRecognizer) {
+        guard let drawnQuad = quadView.quad else {
+            return
+        }
+        
+        guard pan.state != .ended else {
+            self.previousPanPosition = nil
+            self.closestCorner = nil
+            quadView.resetHighlightedCornerViews()
```

Comment:
`self` is not needed

Reply:
The compiler is not happy if I don't have self here, because I declare variables with the same name afterwards.